specs/PushNotifications/PushNotifications-spec.md

```diff
@@ -0,0 +1,400 @@
+        byte[] Payload { get; };
+
+        // Takes a deferral to run under specific modes like low power mode
+        void GetDeferral();
+
+        // Closes the Deferral taken on a Push Trigger. If this is not called by the developer, the deferral will automatically be invoked on destruction.
+        void CompleteDeferral();
+
+        // Subscribe to Cancelled event handler to be signalled when resource policies are no longer true like 30s wallclock timer
+        event Windows.ApplicationModel.Background.BackgroundTaskCanceledEventHandler Canceled;
+
+        // The number of times Resource management suspended the Task
+        UInt32 SuspendedCount { get; };
+    };
+
+    // An abstraction over the inproc activation Registration flow
+    runtimeclass InProcActivatorDetails
+    {
+        // Initialize using a manifest defined activatorId to Activate the InProc COM server
+        InProcActivatorDetails(Guid taskClsid);
+
+        // Initialize an inproc activator for LOW-IL apps like UWP with no COM Server specified in manifest
+        InProcActivatorDetails();
+
+        // The CLSID associated with the Client COM server that reunion will adopt
+        Guid TaskClsid{ get; };
+
+        // The conditions under which Push Triggers would execute
+        Windows.ApplicationModel.Background.IBackgroundCondition Condition;
+
+        // Event handler to subscribe to if the app is in the foreground
+        event Windows.Foundation.EventHandler<PushActivatedEventArgs> PushActivated;
+
+        // The backgroundRegistration that gets populated on successful registration
+        Windows.ApplicationModel.Background.IBackgroundTaskRegistration Registration;
+    };
+
+    enum ChannelStatus
+    {
+        InProgress, // The request is in progress and there is no retry operation
+        InProgressRetry, // The request is in progress and is in a backoff retry state. Check ExtendedError for HRESULT for retryable error.
+        CompletedSuccess, // The request completed successfully
+        CompletedFailure, // The request failed with some critical internal error. Check ExtendedError for HRESULT
+    };
+
+    runtimeclass ChannelResult
+    {
+        ChannelResult(
+            Windows.Networking.PushNotifications.PushNotificationChannel channel,
+            HRESULT extendedError,
+            ChannelStatus status);
+
+        // The Push channel associated with the Result. Null if InProgress or completion failed
+        Windows.Networking.PushNotifications.PushNotificationChannel Channel { get; };
+
+        // More detailed error code in addition to the ChannelStatus state.
+        HRESULT ExtendedError{ get; };
+
+        // The Status of the ChannelComplete operation
+        ChannelStatus Status { get; };
+    };
+
+    static runtimeclass PushManager
+    {
+        // Register an activator using a Details context for inproc activations
+        static void RegisterActivator(InProcActivatorDetails details);
```

Comment:
Needs to be updated to take in a RegistrationType

Reply:
RECOMMEND: `PushNotificationRegistrationKind`